src/Microsoft.AspNetCore.Routing/Template/TemplateBinder.cs

```diff
@@ -72,7 +98,7 @@ public TemplateValuesResult GetValues(RouteValueDictionary ambientValues, RouteV
                 var parameterName = parameter.Name;
 
                 object newParameterValue;
-                var hasNewParameterValue = values.TryGetValue(parameterName, out newParameterValue);
+                var hasNewParameterValue = context.AcceptedValues.TryGetValue(parameterName, out newParameterValue);
```

Comment:
Is this change going to modify the dictionary passed in to routing? We can't do that.